src/macro.jl

```diff
@@ -83,6 +83,7 @@ function floop(ex, simd)
     if ansvar !== :_
         post = vcat(post, ansvar)
     end
+    pre = vcat(pre, something(EXTRA_STATE_VARIABLES[], Union{}[]))
```

Comment:
This PR adds `EXTRA_STATE_VARIABLES[]` to the list of state variables to be used.  This is used via https://github.com/JuliaFolds/FGenerators.jl/pull/7